src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -24,17 +25,33 @@
  */
 public class ModelManager implements Model {
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
-
     private final ContactList contactList;
     private final PriorityTaskList taskList;
     private final UserPrefs userPrefs;
     private final FilteredList<Person> filteredPersons;
-
+    private final UniqueModuleList moduleList;
     /** List of versioned contents */
     private final VersionedContents versionedContents;
 
     /**
-     * Initializes a ModelManager with the given contactList, userPrefs and taskList.
+     * Initializes a ModelManager with the given contactList, userPrefs and a moduleList.
+     */
+    public ModelManager(ReadOnlyContactList contactList, ReadOnlyUserPrefs userPrefs, UniqueModuleList moduleList) {
+        requireAllNonNull(contactList, userPrefs);
```

Comment:
can moduleList be null in this case? If not, it might be a good idea to include it here